Rebus.AdoNet/AdoNetSagaPersisterAdvanced.cs

```diff
@@ -289,22 +289,10 @@ private string GetSagaLockingClause(SqlDialect dialect)
 			return string.Empty;
 		}
 
-		private bool ShouldIndexValue(object value)
-		{
-			if (IndexNullProperties)
-				return true;
-
-			if (value == null) return false;
-			if (value is string) return true;
-			if ((value is IEnumerable) && !(value as IEnumerable).Cast<object>().Any()) return false;
-
-			return true;
-		}
-
 		private IDictionary<string, object> GetCorrelationItems(ISagaData sagaData, IEnumerable<string> sagaDataPropertyPathsToIndex)
 		{
 			return sagaDataPropertyPathsToIndex
-				   .Select(x => new { Key = x, Value = Reflect.Value(sagaData, x) })
```

Comment:
Are we sure 'sagaDataPropertyPathsToIndex' never contains duplicates? Cause if so, the following ToDictionary is going to throw..

Reply:
Before, method was doing Select+Where(w/o discarding anything)+ToDictionary.
Now, w/ this change, ToDictionary+Where(fixed)+ToDictionary.

I think duppes are not present, also I think that raising an exception instead of skipping dupes is a good aproach as correlation key should be unique, no?

Regards